### PR TITLE
Feature/improve near

### PR DESCRIPTION
**Pull request by Luicel**
Improved local distance (both for /near and local chat) to include chained players. Basically Blast did this one tbh.

Also fixed df format in /distance command.

### CODE REVIEW

**Comment by Luicel** [comment]
Potentially need to reverse the set in NearCommand (so closer players are shown first in the list). Will look into this later.

**Comment by qixils** [comment]
> Potentially need to reverse the set

HashSet does not support ordering, you should use LinkedHashSet

also, what are "chained players"? i feel this should be clarified in the hover text

**Comment by Blast-MC** [comment]
You can also use a TreeSet and compare the distance to have the set automatically sorted closest to furthest

**Comment by Blast-MC** [comment]
I updated to use a TreeSet, however, I noticed that it doesn't query who you can see, but rather who the chained players can see. Therefore, you can see vanished staff if 1 staff happens to be unvanished in the range. You'll most likely need to pass the original player in the NearCommand#getNearbyPlayers method and use that in the PlayerUtils#canSee call.

**Comment by Luicel** [comment]
> also, what are "chained players"? i feel this should be clarified in the hover text

Done. 

Chaining is just recursion for the local radius. If a player is 600 blocks away from you but a player is between you at a distance of 300 blocks, the 600 block-away player will be included in /near and local chat. So in local chat players will not seem to be talking to nothing anymore. But yeah, a small description in /near would definitely be nice haha.

Such a small, minuscule thing, but I thought it'd be a nice QoL update. It's not a perfect solution but I think it's better than how it used to be.

> I updated to use a TreeSet, however, I noticed that it doesn't query who you can see, but rather who the chained players can see. Therefore, you can see vanished staff if 1 staff happens to be unvanished in the range. You'll most likely need to pass the original player in the NearCommand#getNearbyPlayers method and use that in the PlayerUtils#canSee call.

Thanks for including the TreeSet haha. 

And done - nice catch. However, the method definition looks a bit messy. Is there a way to improve this?

File: src/main/java/gg/projecteden/nexus/features/commands/NearCommand.java [hunk 1]
```diff
@@ -31,30 +33,60 @@ void run(@Arg(value = "self", permission = "group.staff") Player player) {
...
+
+			for (Player _player2 : nearby) {
+				if (!nearbyPlayers.contains(_player2))
+					getNearbyPlayers(originalPlayer, _player2, nearbyPlayers, includeUnseen);
+			}
+		}
+		return nearbyPlayers;
+	}
+
+	static private long getDistance(@Arg("self") Player player, Player _player) {
```
Comment (GriffinCodes) [inline src/main/java/gg/projecteden/nexus/features/commands/NearCommand.java:89]: Annotation not necessary outside of command framework

**Comment by GriffinCodes** [comment]
> However, the method definition looks a bit messy. Is there a way to improve this?
Cleaned it up for you with a builder pattern

**Comment by Luicel** [comment]
> Cleaned it up for you with a builder pattern

Well damn, thank you haha.

**Pull request merged**